packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Models/Types/ModelTypeProvider.cs

```diff
@@ -49,21 +51,208 @@ protected override PropertyDeclaration[] BuildProperties()
+            {
+                FormattableString binaryDataExtraDescription = CreateBinaryDataExtraDescription(type, serializationFormat);
+                description = $"{description}{binaryDataExtraDescription}";
+            }
+
+            return description;
+        }
+
+        /// <summary>
+        /// This method attempts to retrieve the description for each of the union type items. For items that are lists,
+        /// the description will include details about the element type.
+        /// For items that are literals, the description will include the literal value.
+        /// </summary>
+        /// <param name="unionItems">the list of union type items.</param>
+        /// <returns>A list of FormattableString representing the description of each union item.
+        /// </returns>
+        internal static IReadOnlyList<FormattableString> GetUnionTypesDescriptions(IReadOnlyList<CSharpType> unionItems)
+        {
+            var values = new List<FormattableString>();
+
+            foreach (CSharpType item in unionItems)
+            {
+                FormattableString description;
+
+                if (item.IsLiteral && item.Literal?.Value != null)
+                {
+                    var literalValue = item.Literal.Value.Value;
+                    if (item.FrameworkType == typeof(string))
+                    {
+                        description = $"<description>{literalValue:L}</description>";
+                    }
+                    else
+                    {
+                        description = $"<description>{literalValue}</description>";
+                    }
+                }
+                else
+                {
+                    description = $"<description>{item:C}</description>";
+                }
+
+                values.Add(description);
+            }
+
+            return values.Distinct().ToList();
+        }
+
+        private FormattableString ConstructBinaryDataDescription(string typeSpecificDesc, SerializationFormat serializationFormat, IReadOnlyList<FormattableString> unionTypeDescriptions)
+        {
+            FormattableString unionTypesAdditionalDescription = $"";
+
+            if (unionTypeDescriptions.Count > 0)
+            {
+                unionTypesAdditionalDescription = $"\n<remarks>\nSupported types:\n<list type=\"bullet\">\n";
+                foreach (FormattableString unionTypeDescription in unionTypeDescriptions)
+                {
+                    unionTypesAdditionalDescription = $"{unionTypesAdditionalDescription}<item>\n{unionTypeDescription}\n</item>\n";
+                }
+                unionTypesAdditionalDescription = $"{unionTypesAdditionalDescription}</list>\n</remarks>";
+            }
+            switch (serializationFormat)
+            {
+                case SerializationFormat.Bytes_Base64Url:
+                // cSpell:disable
+                case SerializationFormat.Bytes_Base64:
+                    return $@"
+<para>
```

Comment:
Can we add an example that exercises this option as well in the test project?

Reply:
done.